src/infra/azure/functions/validate-document-upload.ts

```diff
@@ -11,38 +11,71 @@ import { validate } from "@pagopa/handler-kit/lib/validation";
 import * as E from "fp-ts/Either";
 import * as A from "fp-ts/Array";
 
-import { SignatureRequestId } from "../../../signature-request/signature-request";
-import { SubscriptionId } from "../../../signature-request/subscription";
-import { DocumentId } from "../../../signature-request/document";
+import {
+  DocumentId,
+  documentNotFoundError,
+} from "../../../signature-request/document";
 import {
   makeChangeDocumentStatus,
   makeValidateDocument,
-  ValidateDocumentPayload,
 } from "../../../app/use-cases/validate-document";
 import {
   getSignatureRequest,
   upsertSignatureRequest,
 } from "../cosmos/signature-request";
 
-import { config } from "../../../app/config";
+import { Config, config } from "../../../app/config";
 
 import { createContainerClient } from "../storage/client";
-import { makeIsDocumentUploaded } from "../storage/document";
+import {
+  makeDeleteDocumentUploaded,
+  makeIsDocumentUploaded,
+  makeMoveUploadDocument,
+} from "../storage/document";
+import {
+  getUploadDocument,
+  upsertUploadDocument,
+} from "../cosmos/upload-document";
+import { makeDeleteUploadedDocument } from "../../../app/use-cases/delete-uploaded-document";
+
+const issuerContainerClient = (cfg: Config) =>
+  createContainerClient(
+    cfg.storage.connectionString,
+    cfg.storage.issuerBlobContainerName
+  );
+
+const issuerValidatedContainerClient = (cfg: Config) =>
+  createContainerClient(
+    cfg.storage.connectionString,
+    cfg.storage.issuerValidatedBlobContainerName
+  );
 
+const unableToConnectError = new Error(
+  "Unable to connect to Blob Storage Service"
+);
 const isDocumentUploadedToBlobStorage = pipe(
   config,
-  E.map((config) =>
-    createContainerClient(
-      config.storage.connectionString,
-      config.storage.issuerBlobContainerName
-    )
-  ),
+  E.map(issuerContainerClient),
   E.map(makeIsDocumentUploaded),
+  E.getOrElse(() => (_) => TE.left(unableToConnectError))
+);
+
+const moveDocumentUrlToValidatedBlobStorage = pipe(
+  config,
+  E.map(issuerValidatedContainerClient),
+  E.map(makeMoveUploadDocument),
   E.getOrElse(
-    () => (_) => TE.left(new Error("Unable to connect to Blob Storage Service"))
+    () => (_sourceDocumentUrl, _documentId) => TE.left(unableToConnectError)
```

Comment:
```suggestion
    () => () => TE.left(unableToConnectError)
```

is this right? (to return the method instead of executing it)

Reply:
Yes, is right. This function is only to access the configuration and set the desired parameters, then the method is returned and subsequently recalled.